hermes/build.rs

```diff
@@ -84,11 +86,30 @@ fn main() {
         .arg("src/network/p2p.go")
         .arg("src/network/p2p.pb.go");
 
+    // Cross-compile the Go binary based on the Rust target architecture
+    match &*rust_target_arch {
+        "x86_64" => {
+            // CGO_ENABLED required for building amd64 on mac os
+            cmd.env("GOARCH", "amd64").env("CGO_ENABLED", "1");
+        }
+        "aarch64" => {
+            cmd.env("GOARCH", "arm64");
+        }
+        // Add other target architectures as needed
+        _ => {
+            panic!("Unsupported target architecture: {}", rust_target_arch);
+        }
```

Comment:
Can you remove this to allow building in those targets as well? As long as rust and go use the same architecture this code should compile.